include/depthai/depthai_wrapper.hpp

```diff
@@ -0,0 +1,64 @@
+#include "depthai/device.hpp"
+#include "nlohmann/json.hpp"
+#include <opencv2/core/core.hpp>
+#include <opencv2/opencv.hpp>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace DepthAI {
+
+using CV_mat_ptr = std::shared_ptr<cv::Mat>; // shared ptr for cv::Mat
+using PacketsTuple = std::tuple<
+    std::list<std::shared_ptr<NNetPacket>>,
+    std::list<std::shared_ptr<HostDataPacket>>>; // tuple of data packets on which Depthai is
+// publishing the streams
+
+/**  DepthAI:
+ *   This is a wrapper on the host side of the device which 
+ *   helps in initializing device, creating pipeline
+ *   and fetching the frames as cv::Mat. 
+ */
+
+class DepthAI : public Device {
+
+public:
+    /**  Constructor:
+     *   initializing device and creating pipeline
+     *  @param usb_device (std::string): Provide the path of the device
+     *  or pass empty string to choose the default device.
+     *  @param config_file (std::string): Provides the json file which is
+     *  used by the Depthai to configure the device
+     *  @param usb2_mode (bool): set to true to connect over usb2
+     */
+    DepthAI(const std::string& usb_device, const std::string& config_file, bool usb2_mode = false);
+
+    /* API to stream output frames from device.
+     * AI data and metadata is WIP
+     */
+    void get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams);
+
+    /* Destructor */
+    ~DepthAI() = default;
+    static const std::unordered_map<int, int> width_map_; // mapping of width with height
```

Comment:
I don't understand the comment. Does it mean that each width corresponds to a height?. Then, please call this map `width_to_height_map_`.